src/txn/txn.c

```diff
@@ -830,15 +854,67 @@ __txn_search_prepared_op(
 static int
 __txn_resolve_prepared_op(WT_SESSION_IMPL *session, WT_TXN_OP *op, bool commit, WT_CURSOR **cursorp)
 {
+    WT_CURSOR *hs_cursor;
+    WT_DECL_RET;
     WT_TXN *txn;
-    WT_UPDATE *upd;
-    bool resolved;
+    WT_UPDATE *fix_upd, *upd;
+    uint32_t hs_btree_id, session_flags;
+    bool is_owner, upd_appended;
 
+    fix_upd = NULL;
+    hs_cursor = NULL;
     txn = session->txn;
-    resolved = false;
+    session_flags = 0;
+    upd_appended = is_owner = false;
 
     WT_RET(__txn_search_prepared_op(session, op, cursorp, &upd));
 
+    /*
+     * Aborted updates can exist in the update chain of our txn. Generally this will occur due to a
+     * reserved update. As such we should skip over these updates.
+     */
+    for (; upd->txnid == WT_TXN_ABORTED; upd = upd->next)
+        ;
+
+    /*
+     * The head of the update chain is not a prepared update, which means all the prepared updates
+     * of the key are resolved.
+     */
+    if (upd->prepare_state != WT_PREPARE_INPROGRESS)
```

Comment:
No, this is just to prevent us from going into the following code again if we have multiple prepared updates from the same transaction. They get resolved together however we may enter this function for the same key multiple times because each update is a different operation in the transaction.